duouniversal/client.go

```diff
@@ -218,28 +218,60 @@ func newStrictTLSTransport() *http.Transport {
 	}
 }
 
-func NewClient(clientId, clientSecret, apiHost, redirectUri string) (*Client, error) {
-	return NewClientDuoCodeAttribute(clientId, clientSecret, apiHost, redirectUri, true)
+func NewClient(clientId, clientSecret, apiHost, redirectUri string, opts ...clientOptions) (*Client, error) {
+	opts = append(opts, UseDuoCodeAttribute())
+	return newClient(clientId, clientSecret, apiHost, redirectUri, opts...)
 }
 
 // Creates a new Client with the ability to turn off use_duo_code_attribute
 func NewClientDuoCodeAttribute(clientId, clientSecret, apiHost, redirectUri string, useDuoCodeAttribute bool) (*Client, error) {
-	if len(clientId) != clientIdLength {
-		return nil, fmt.Errorf(clientIdError)
-	} else if len(clientSecret) != clientSecretLength {
-		return nil, fmt.Errorf(clientSecretError)
-	}
-
-	return &Client{
-		clientId:            clientId,
-		clientSecret:        clientSecret,
-		apiHost:             apiHost,
-		redirectUri:         redirectUri,
-		useDuoCodeAttribute: useDuoCodeAttribute,
-		duoHttpClient: &http.Client{
-			Transport: newStrictTLSTransport(),
-		},
-	}, nil
+    if useDuoCodeAttribute {
+        return newClient(clientId, clientSecret, apiHost, redirectUri, UseDuoCodeAttribute())
+    }
+    return newClient(clientId, clientSecret, apiHost, redirectUri)
+}
+
+// Creates a new Client with the ability to turn off use_duo_code_attribute
+func newClient(clientId, clientSecret, apiHost, redirectUri string, opts ...clientOptions) (*Client, error) {
+    if len(clientId) != clientIdLength {
+        return nil, fmt.Errorf(clientIdError)
+    }
+    if len(clientSecret) != clientSecretLength {
+        return nil, fmt.Errorf(clientSecretError)
+    }
+
+    c := &Client{
+        clientId:            clientId,
+        clientSecret:        clientSecret,
+        apiHost:             apiHost,
+        redirectUri:         redirectUri,
+        useDuoCodeAttribute: false,
+        duoHttpClient: &http.Client{
+            Transport: newStrictTLSTransport(),
+        },
+    }
+
+    for _, option := range opts {
+        option(c)
```

Comment:
You are correct it would be swallowed up with the current implementation of the options. I would argue that options should avoid returning errors, but if they need to this can be changed to propagate an error up through newClient.

Reply:
I tend to agree with you :)  I don't think we need to make any changes right now, I just wanted to make sure I was understanding how this would work.